### PR TITLE
Throw SysError if the buildHook file does not exist

### PR DESCRIPTION
**Pull request by matthiasbeyer**
A follow-up to #957 .

---

**Please note**: This is the very first piece of C++ code I write outside of university. I hope the place is appropriate and everything.

I did not yet test this, I first wanted to get feedback whether this change is appropriate at all...

### CODE REVIEW

**Comment by bjornfor** [comment]
If this goes in (I have no say in the matter), can you make the error message say which path it tried?

**Comment by matthiasbeyer** [comment]
@bjornfor I tried to do that in the latest commit, thanks for suggesting it!

**Comment by domenkozar** [comment]
cc @edolstra

File: src/libstore/build.cc [hunk 1]
```diff
@@ -596,6 +596,9 @@ HookInstance::HookInstance()
     if (string(buildHook, 0, 1) != "/") buildHook = settings.nixLibexecDir + "/nix/" + buildHook;
     buildHook = canonPath(buildHook);
 
+    if (!pathExists(buildHook))
+        throw SysError(format("NIX_BUILD_HOOK '%1%' does not exist" % buildHook));
```
Comment (edolstra) [inline src/libstore/build.cc]: This won't compile. The `% buildHook` needs to be after the first `)`.

**Pull request closed without merging**